driver-core/src/main/com/mongodb/ReadConcern.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb;
+
+import static com.mongodb.assertions.Assertions.notNull;
+import static java.lang.String.format;
+
+/**
+ * The readConcern option allows clients to choose a level of isolation for their reads.
+ *
+ * The WiredTiger storage engine, introduces the readConcern option for replica sets and replica set shards.  Allowing clients to choose a
+ * level of isolation for their reads.
+ *
+ * @mongodb.server.release 3.2
+ * @since 3.2
+ */
+public enum ReadConcern {
```

Comment:
I can wrap it I originally did but it seemed cumbersome - especially as I misread the spec and it could also take a string.

I then migrated to just using a string - as again I took the spec to literally, as it said I couldn't error on any user provided value for future proofing making enums useless. Having clarified that I went back to enums - but nesting them felt made it awkward.

I'll add some static helpers to mean the same as the enum levels and then nest the level, its easy enough and users wont notice the semantic difference until extra options become available.